From 19832977732edbfa4be2d5a38ac0c093e6b43a2d Mon Sep 17 00:00:00 2001
From: Yi Gu <yigu@chromium.org>
Date: Mon, 11 Jul 2022 22:32:39 +0000
Subject: [PATCH 075/110] [Sheriff] Disable flaky MachOImageAnnotationsReader
 tests for M102

The tests have been flaky:
https://test-results.appspot.com/dashboards/flakiness_dashboard.html#testType=crashpad_tests%20(with%20patch)&tests=MachOImageAnnotationsReader

(cherry picked from commit b867f2c69834ea0a5fcdee093d03b7be817ba03c)

Bug: 1334418
Change-Id: I7a887a4af27fc02056ff5c2b68fc912056d9e7fe
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3703673
Auto-Submit: Yi Gu <yigu@chromium.org>
Owners-Override: Yi Gu <yigu@chromium.org>
Commit-Queue: Anthony Cui <cuianthony@chromium.org>
Reviewed-by: Anthony Cui <cuianthony@chromium.org>
Cr-Original-Commit-Position: refs/heads/main@{#1014152}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3754553
Reviewed-by: Sky Malice <skym@chromium.org>
Commit-Queue: Tomasz Wiszkowski <ender@google.com>
Auto-Submit: Tomasz Wiszkowski <ender@google.com>
Owners-Override: Tomasz Wiszkowski <ender@google.com>
Cr-Commit-Position: refs/branch-heads/5005@{#1234}
Cr-Branched-From: 5b4d9450fee01f821b6400e947b3839727643a71-refs/heads/main@{#992738}
---
 .../mac/mach_o_image_annotations_reader_test.cc     | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/third_party/crashpad/crashpad/snapshot/mac/mach_o_image_annotations_reader_test.cc b/third_party/crashpad/crashpad/snapshot/mac/mach_o_image_annotations_reader_test.cc
index 74a21be670ce..8b7e3ff15aae 100644
--- a/third_party/crashpad/crashpad/snapshot/mac/mach_o_image_annotations_reader_test.cc
+++ b/third_party/crashpad/crashpad/snapshot/mac/mach_o_image_annotations_reader_test.cc
@@ -460,19 +460,16 @@ TEST(MachOImageAnnotationsReader, CrashAbort) {
   test_mach_o_image_annotations_reader.Run();
 }
 
-#if defined(ADDRESS_SANITIZER)
-// https://crbug.com/844396
-#define MAYBE_CrashModuleInitialization DISABLED_CrashModuleInitialization
-#else
-#define MAYBE_CrashModuleInitialization CrashModuleInitialization
-#endif
-TEST(MachOImageAnnotationsReader, MAYBE_CrashModuleInitialization) {
+// Flaky on ASAN https://crbug.com/844396
+// Flaky in general https://crbug.com/1334418
+TEST(MachOImageAnnotationsReader, DISABLED_CrashModuleInitialization) {
   TestMachOImageAnnotationsReader test_mach_o_image_annotations_reader(
       TestMachOImageAnnotationsReader::kCrashModuleInitialization);
   test_mach_o_image_annotations_reader.Run();
 }
 
-TEST(MachOImageAnnotationsReader, CrashDyld) {
+// Flaky in general https://crbug.com/1334418
+TEST(MachOImageAnnotationsReader, DISABLED_CrashDyld) {
   TestMachOImageAnnotationsReader test_mach_o_image_annotations_reader(
       TestMachOImageAnnotationsReader::kCrashDyld);
   test_mach_o_image_annotations_reader.Run();
-- 
2.38.1

